### ISSUE TITLE
Vanish and Shadowmeld mechanics - not able to mitigate abilities

### ISSUE THREAD

**Issue by ujellyx** [SunwellTracker/issues#1505]
**Decription:** 
So, this has already been reported (#1190), but OP didn't look up any proof (videos or else), so I'll report it again - this time with some proof.
Rogues or Nightelves should be able to vanish/shadowmeld projectiles/spells that are already in air or abilities, that are cast just as the vanish/shadowmeld is being used. Right now, this doesn't work, all the abilities still hit.

**How it works:** 
Projectiles already in air or abilities cast right when vanish/shadowmeld is cast, still hit (do damage or cc).

**How it should work:** 
Projectiles already in air or abilities cast right when vanish/shadowmeld is cast (due to short immunity time), should be dodged. (except Engineering glove enchant)

**Source (you should point out proofs of your report, please give us some source):** 
From retail:
http://www.warcraftmovies.com/movieview.php?id=110241
4:15 vanishes a Shadowfury
4:19 vanishes a Deathcoil, completely mitigating it
4:50 vanishes a Deathcoil again
6:51 same as above
17:51 vanishes a Blind from the opposite rogue
18:07 opposite rogue vanishes players Blind

From Arena-Tournament/Warmane (since they have everything right PvP mechanic wise):
https://www.youtube.com/watch?v=1wikzG0orOs
0:43 Shadowmelding the Shadowfury
5:58 Shadowmelding enemy's Kidney Shot
7:47 Shadowmelding Deathcoil and Incinerate

If you need more videos, feel free to ask, I'm sure I can find many many more examples.

This is on wowwiki aswell:
"Back in the days Vanish allowed avoiding getting hit by mid-air spells. Later on you could even negate instant cast spell effects due to short amount of time of immunity Vanish provided, which was added to the game because of rogue players complains about Vanish being randomly broken. This was changed in early Cataclysm after Vanish was reworked into providing unbreakable 3 second Stealth."
http://wowwiki.wikia.com/wiki/Vanish
This states exactly what I described in the report itself and was changed in Cataclysm.

**TL;DR:** Make it possible to dodge incoming projectiles mid-air and add a short immunity after vanish is used.

**Comment by Nadiirr** [comment]
Thank you for the very detailed post.

**Comment by lukaasm** [comment]
In most cases vanish/shadowmeld should now ignore already flying spells + together with introduced CU_FAKE_DELAY, it should be usable against most instant CC spells if timed right.

**Comment by ujellyx** [comment]
Does this also count for projectiles (hunter-shots + autoshots)? Is the small immunity window also implemented?
